library/ssl_tls13_server.c

```diff
@@ -1072,6 +1236,7 @@ int mbedtls_ssl_tls13_handshake_server_step( mbedtls_ssl_context *ssl )
     {
         /* start state */
         case MBEDTLS_SSL_HELLO_REQUEST:
+            ssl->handshake->hello_retry_request_count = 0;
```

Comment:
I'd suggest to just remove this. This is done by `ssl_handshake_params_init` and there is no HelloRequest message in TLS 1.3. When we will support version negotiation on the server side, the handling of this state will be moved to ssl_tls.c like in the client case.